astropy/io/ascii/cds.py

```diff
@@ -608,44 +620,52 @@ def write(self, lines):
                             coord_col = Column(list(coord), name=name, unit=coord_unit,
                                                description=descrip)
                         # Set default number of digits after decimal point for the
-                        # second values.
+                        # second values, and deg-min to (signed) 2-digit zero-padded integer.
                         if name in ['RAs', 'DEs']:
-                            coord_col.format = '.12f'
+                            coord_col.format = '015.12f'
+                        elif name == 'RAh':
+                            coord_col.format = '2d'
+                        elif name == 'DEd':
+                            coord_col.format = '+03d'
+                        elif name.startswith(('RA', 'DE')):
+                            coord_col.format = '02d'
                         self.cols.append(coord_col)
 
                 # For all other coordinate types, simply divide into two columns
                 # for latitude and longitude resp. with the unit used been as it is.
+                # Galactic coordinates.
+                elif col.name == 'galactic' and 'GLAT' not in self.colnames:
```

Comment:
The primary incentive here was to check if any prior coordinate column of the same kind had already been expanded, in which case there would always been both `'GLAT'` and `'GLON'` columns; analogously for '`RAh'` etc. But of course if the original table perchance already had a column named 'GLON', that would of course be missed and a duplicate column created like
```
 Bytes Format Units  Label     Explanations
--------------------------------------------------------------------------------
 1- 7  A7     ---    name    Description of name  
 9-15  F7.3   ---    GLON    Description of GLON  
17-32  F16.12 deg    GLON    Galactic Longitude   
34-49  F16.12 deg    GLAT    Galactic Latitude    
51-64  A14    ---    coord2  Description of coord2
--------------------------------------------------------------------------------
Notes:
--------------------------------------------------------------------------------
HD4760    6.402  12.416250000000   6.402222000000 12.4163 6.407 
HD81809 -61.660 330.556444000000 -61.659611110000 330.564 -61.66
```
Might say that it's up to the user to avoid such ambiguities, but perhaps it's better to check `lon` and `lat` in a consistent manner.

Reply:
Rewrote the LON/LAT block in a more general (and hopefully also easier to extend) manner, which checks for all relevant columns.